### PR TITLE
Demonic Government

### PR DESCRIPTION
**Pull request by zumbak04**
## Changelog:
- Added Demonic Government:
  - No vassal contracts, taxes and levies based on opinion.
  - Only rulers of the same religious family can inherit.
  - Dark Authority instead of Crown Authority:
    - Frenzy decays over time so to be efficient they should do their mechanics.
    - High Frenzy gives you movement speed, dread and tyranny bonuses.
  - Get Frenzy on executions.
  - Get Frenzy on torturing.
  - Attracts Demon Knights and Guests with high Skills.
- Assigned modifiers/effects to Demonic races: satyrs, observers, etc.

# Developer changelog:
- Added debug decisions to lose and give Dark Frenzy.

# Tasks:
- [x] Add debug decisions
- [x] Mention unique bonuses in government concept.

# How to test:
Test gameplay features, tell if they look balanced.

In the Debug mode, right-click Demonic government rulers to Give and Lose Dark Frenzy fast.

<details>
<summary>Click to expand</summary>

image

</details>

@Warcraft-GoA-Development-Team/english-proofreaders open the [Page](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2/pull/592/files?file-filters%5B%5D=.yml), scroll to the bottom to the `.yml` files, check them, suggest changes. If you forgot, here are the [Suggestion Guidelines](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2/wiki/How-to-be-Tester-(For-CK3)#reviewing-code).

### CODE REVIEW

File: localization/english/wc_government_l_english.yml [hunk 1]
```diff
@@ -10,3 +10,13 @@
  government_dark_frenzy_from_conversions:0 "Gets [dark_frenzy|E] from [convert_county|E] and [promote_culture|E]"
  
  government_resurrect_knights:0 "Killed enemy [knights|E] become [GetTrait('being_undead').GetName( GetNullCharacter )] and join $necro_government_adjective$ [court|E] #F (high [dark_authority|E] increases the chance)#!"
+ demonic_government:0 "Demonic"
+ demonic_government_adjective:0 "Demonic"
+ demonic_government_realm:0 "Demonic Realm"
+ demonic_government_desc:0 "\n$game_concept_demonic_government_desc$\n\n#F Demonic hordes that came out of the Twisting Nether to bring havoc. Their immortal souls are hungry for pain and tortures caused upon mortal civilizations.#!"
```
Comment (LordSilvermane) [inline localization/english/wc_government_l_english.yml:16]:
```suggestion
 demonic_government_desc:0 "\n$game_concept_demonic_government_desc$\n\n#F Demonic hordes that have come out of the Twisting Nether to wreak havoc. Their immortal souls are always hungry for pain and suffering inflicted upon mortal civilizations.#!"
```

**Review by LordSilvermane — COMMENTED** [review]
One suggestion, otherwise looks good.

**Review by CantTouchThisJinTao — CHANGES_REQUESTED** [review]
Most seems to be working as intended, however:

- Torturing doesn't result in gaining frenzy (I assume that's what was meant and not torturers - either way torturing and the torturer trait don't give frenzy)
- Shivarran have the nathrezim trait (is this intended).
- Eredar do not have the demon trait (or have modifiers added to their trait - I guess this is intended though since draenei have the same trait).

**Review by LordSilvermane — APPROVED** [review]
LGTM!

**Comment by zumbak04** [comment]
> * Torturing doesn't result in gaining frenzy (I assume that's what was meant and not torturers - either way torturing and the torturer trait don't give frenzy)

I can't confirm.

<details>
<summary>Click to expand</summary>

image

</details>


> * Shivarran have the nathrezim trait (is this intended).

Fixed in 217c61fde67a1a6a2d433ae69a87ab7a94d37a68.

**Comment by zumbak04** [comment]
> I can't confirm.

But may be fixed in 6b4db066020baade95370247fdee79fef98a9e92. So worth testing on your side again.

**Comment by CantTouchThisJinTao** [comment]
> > * Torturing doesn't result in gaining frenzy (I assume that's what was meant and not torturers - either way torturing and the torturer trait don't give frenzy)
> 
> I can't confirm.
> 
> Click to expand
> > * Shivarran have the nathrezim trait (is this intended).
> 
> Fixed in [217c61f](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2/commit/217c61fde67a1a6a2d433ae69a87ab7a94d37a68).

Yeah torture seems to give dark frenzy now, shivarran have their trait as well.

I'll approve now then.

**Review by CantTouchThisJinTao — APPROVED** [review]

**Comment by zumbak04** [comment]
> * Eredar do not have the demon trait (or have modifiers added to their trait - I guess this is intended though since draenei have the same trait).

Fixed in 18c94fd54027eeff36f2b94d54673c3663f40b74

**Comment by CantTouchThisJinTao** [comment]
> > * Eredar do not have the demon trait (or have modifiers added to their trait - I guess this is intended though since draenei have the same trait).
> 
> Fixed in [18c94fd](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2/commit/18c94fd54027eeff36f2b94d54673c3663f40b74)

Can confirm starting as Demonic Eredar via ruler designer gives the Demon trait, none of the already existing one's have it though.

**Review by arithon — APPROVED** [review]
loc LGTM

**Comment by zumbak04** [comment]
> Can confirm starting as Demonic Eredar via ruler designer gives the Demon trait, none of the already existing one's have it though.

Fixed in a1aca54691b220ad7b4e9982e9c6b3fb49ccbfdc.

**Review by CantTouchThisJinTao — APPROVED** [review]
Everything seems to be working now

**Review by TheGroxEmpire — COMMENTED** [review]
- Torturing non ruler is still not rewarding Dark Frenzy:
<Details>

image
</details>

> If you want combat or army bonuses, I can make the demonic army stronger instead of cheaper. May pick one of these bonuses.

I'm fine with that. I think aggressive attackers (+25% Enemy Fatal Casualties) is thematic and doesn't tip the balance too much.

**Comment by zumbak04** [comment]
> Torturing non ruler is still not rewarding Dark Frenzy:

Ruler or close relative of a ruler (spouse, child).

**Comment by zumbak04** [comment]
> I'm fine with that. I think aggressive attackers (+25% Enemy Fatal Casualties) is thematic and doesn't tip the balance too much.

Done in c62d8d9ad12de023a9fcab8eb9bf05fa3bba54f5.

**Review by TheGroxEmpire — APPROVED** [review]
Everything seems to be working as intended. My suggestion was also implemented.

**Pull request merged**